dftimewolf/lib/collectors/stackdriver.py

```diff
@@ -0,0 +1,102 @@
+from oauth2client.client import ApplicationDefaultCredentialsError
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import StackdriverLogs
+# Need to register with in the protobuf registry.
+from dftimewolf.lib.collectors import audit_log_pb2 as _
+
+
+# Monkey patching the ProtobufEntry because of various issues, notably
+# https://github.com/googleapis/google-cloud-python/issues/7918
+def custom_to_api_repr(self):
+  """API repr (JSON format) for entry."""
+  info = super(logging.entries.ProtobufEntry, self).to_api_repr()
+  info['protoPayload'] = self.payload
+  return info
+
+
+logging.entries.ProtobufEntry.to_api_repr = custom_to_api_repr
+
+
+class StackdriverLogsCollector(module.BaseModule):
+  """Collector for Stackdriver logs."""
+
+  def cleanup(self):
+    pass
+
+  def __init__(self, state):
+    """Initializes a Stackdriver logs collector."""
+    super(StackdriverLogsCollector, self).__init__(state)
+    self._project_name = None
+    self._filter_expression = None
+    self._output_path = tempfile.mkdtemp()
+
+  # pylint: disable=arguments-differ
+  def setup(self, project_name, filter_expression):
+    """Sets up a a Stackdriver logs collector.
+
+    Args:
+      project_name (str): name of the project to fetch logs from.
+      filter_expression (str): Stackdriver advanced logs filter expression.
+    """
+    self._project_name = project_name
+    self._filter_expression = filter_expression
+
+  def process(self):
+    """Copies logs from a cloud project."""
+    descending = logging.DESCENDING
+
+    output_path = os.path.join(self._output_path, 'stackdriver.json')
+    output_file = open(output_path, 'w')
+
+    try:
+      logging_client = logging.Client()
+      for entry in logging_client.list_entries(
+          order_by=descending, projects=[self._project_name],
+          filter_=self._filter_expression):
+
+        log_dict = entry.to_api_repr()
+        output_file.write(json.dumps(log_dict))
+        output_file.write('\n')
+
+    except AccessTokenRefreshError as exception:
+      self.state.add_error(
+          'Something is wrong with your gcloud access token.')
+      self.state.add_error(exception, critical=True)
+
+    except (ApplicationDefaultCredentialsError,
+            google_auth_exceptions.DefaultCredentialsError) as exception:
+      self.state.add_error(
+          'Something is wrong with your Application Default Credentials. '
+          'Try running:\n $ gcloud auth application-default login')
+      self.state.add_error(exception, critical=True)
+
+    except HttpError as exception:
+      if exception.resp.status == 403:
+        self.state.add_error(
+            'Make sure you have the appropriate permissions on the project')
+      if exception.resp.status == 404:
+        self.state.add_error(
+            'GCP resource not found. Maybe a typo in the project name?')
+      self.state.add_error(exception, critical=True)
+
+    print(output_path)
```

Comment:
If this is not a debug print, please add a message so the user knows what's going on